analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -644,4 +650,130 @@ private void writeRepositorySettingsFile(Map<ArtifactRepository, Path> repositor
         return allDependencies;
     }
 
+    private void writeReport(Path outputDir, String filename, String text) throws ManipulationException {
+        final Path reportFile = outputDir.resolve(filename);
+
+        try {
+            logger.debug("Writing to file {}", reportFile);
+            Files.createDirectories(outputDir);
+            FileUtils.writeStringToFile(reportFile.toFile(), text, StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new ManipulationException("Unable to write " + reportFile, e);
+        }
+    }
+
+    private void processAlignmentReport(Project project, Configuration configuration, ManipulationCache cache,
+            Set<ProjectVersionRef> nonAligned) throws ManipulationException, IOException {
+        final ManipulationModel alignmentModel = cache.getModel();
+        final String originalGa = alignmentModel.getGroup() + ":" + alignmentModel.getName();
+        final StringBuilder builder = new StringBuilder(500);
+        append(builder, "------------------- project {}", originalGa);
+        final String originalVersion = alignmentModel.getOriginalVersion();
+        final String newVersion = alignmentModel.getVersion();
+        final PME jsonReport = new PME();
+        final List<ModulesItem> modules = jsonReport.getModules();
+        final ModulesItem rootModule = new ModulesItem();
+        final String gav = originalGa + ":" + newVersion;
+        final ProjectVersionRef pvr = SimpleProjectVersionRef.parse(gav);
+        rootModule.getGav().setPVR(pvr);
+        modules.add(rootModule);
```

Comment:
Is this necessary? It must also set this same GAV later around line 720.

Reply:
In my previous analysis, before you force pushed,  I had diagnosed which was the problem. I'm on phone now so am not sure

Reply:
> In my previous analysis, before you force pushed, I had diagnosed which was the problem. I'm on phone now so am not sure

You meant this one?

> I think the problem is in if (!originalVersion.equals(newVersion)) { block - you are adding an extra modulesitem in there. Further, I am not convinced on this version check - surely the execution root should always be added ?

I'll look at the code again. I may have missed the fact that these were two different sentences talking about two different things.

Also in response to:

> The first GAV looks wrong to me? Does PME do this? In fact, could that JSON even be read back in - as it doesn't look like it conforms?

The JSON should get read in during the tests, so it does parse, at least. I think it's the test that's wrong right now, and should fail, then, once this is fixed and itself need to be fixed.